src/Meziantou.Analyzer/Rules/UseShellExecuteAnalyzer.cs

```diff
@@ -0,0 +1,147 @@
+﻿using System.Collections.Immutable;
+using System.Linq;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Operations;
+
+namespace Meziantou.Analyzer.Rules;
+
+[DiagnosticAnalyzer(LanguageNames.CSharp)]
+public sealed class UseShellExecuteAnalyzer : DiagnosticAnalyzer
+{
+    private static readonly DiagnosticDescriptor UseShellExecuteMustBeExplicitlySet = new(
+        RuleIdentifiers.UseShellExecuteMustBeSet,
+        title: "UseShellExecute must be explicitly set",
+        messageFormat: "UseShellExecute must be explicitly set when initializing a ProcessStartInfo",
+        RuleCategories.Usage,
+        DiagnosticSeverity.Info,
+        isEnabledByDefault: false,
+        description: "",
+        helpLinkUri: RuleIdentifiers.GetHelpUri(RuleIdentifiers.UseShellExecuteMustBeSet));
+
+    private static readonly DiagnosticDescriptor UseProcessStartOverload = new(
```

Comment:
I think this rule should be disable by default, similar to `UseShellExecuteMustBeExplicitlySet`